wagtail/api/v2/tests/test_pages.py

```diff
@@ -30,6 +33,19 @@ def get_total_page_count():
     )
 
 
+class Test10411APIViewSet(PagesAPIViewSet):
+    meta_fields = []
+
+
+api_router = WagtailAPIRouter("api_v2_test_pages")
+api_router.register_endpoint("issue_10411", Test10411APIViewSet)
```

Comment:
Thanks!

> Simply importing wagtail.test.urls.api_router into `wagtail/api/v2/tests/test_pages.py` and registering a new endpoint on it doesn't work, probably because it has already generated its URL patterns into `wagtail.test.urls.urlpatterns`.

I was thinking the other way around: `wagtail.test.urls` imports `Test10411APIViewSet` from `wagtail.api.v2.tests.test_pages`, and do `api_router.register_endpoint("issue_10411", Test10411APIViewSet)` using the existing `WagtailAPIRouter` instance.

And if it looks ugly, we could probably move `Test10411APIViewSet` to a new `wagtail.api.v2.tests.views` module, which would be where `wagtail.test.urls` imports the `Test10411APIViewSet`.

I haven't tested it though, does that not work?

Reply:
> I haven't tested it though, does that not work?

It does, and makes for an even cleaner diff: https://github.com/wagtail/wagtail/compare/3da4b686eb...mgax:wagtail:refactor-test-for-10411. My bad, I thought the goal was to completely isolate the test logic, within `wagtail.api.v2.tests`.

Ready for another review.

Reply:
Do you want me to squash the commits? I noticed https://github.com/wagtail/wagtail/pull/10638 was merged with the "rebase and merge" strategy which would not be helpful here.

Reply:
Awesome, this is what I had in mind. Thanks!

> Do you want me to squash the commits?

No need, I can do it as I merge 😄